lint-rules/src/main/java/com/duckduckgo/lint/strings/MissingInstructionDetector.kt

```diff
@@ -0,0 +1,116 @@
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.ResourceXmlDetector
+import com.android.tools.lint.detector.api.Scope
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.XmlContext
+import com.android.tools.lint.detector.api.XmlScanner
+import org.w3c.dom.Attr
+import org.w3c.dom.Document
+import org.w3c.dom.Node
+
+@Suppress("UnstableApiUsage")
+class MissingInstructionDetector : ResourceXmlDetector(), XmlScanner {
+
+    private fun ignoreFile(context: Context) =
+        context.file.name.startsWith("donottranslate") ||
+            ResourceUsageModel.isAnalyticsFile(context.file) ||
+            !context.project.reportIssues
+
+
+    override fun visitDocument(context: XmlContext, document: Document) {
+
+        if (ignoreFile(context)) {
+            return
+        }
+
+        val root = document.documentElement
+        val translatable: Attr? = root?.getAttributeNode(SdkConstants.ATTR_TRANSLATABLE)
+        if (translatable?.value != SdkConstants.VALUE_FALSE) {
+            if (context.file.parentFile?.name == SdkConstants.FD_RES_VALUES) {
+                visitNode(context, document)
+            }
+            return
+        }
+
+    }
+
+    private fun visitNode(context: XmlContext, node: Node, parentHasInstruction: Boolean = false) {
+        var parentInstruction = parentHasInstruction
+        val nodeType = node.nodeType
+        if (nodeType == Node.TEXT_NODE) {
+            val text = node.nodeValue
+            val instruction = node.parentNode.attributes.getNamedItem("instruction")
+            val translatable = node.parentNode.attributes.getNamedItem("translatable")
+            val translatableValue = (translatable as Attr?)?.value
+
+            if (translatableValue != "false" && PLACEHOLDERS.firstOrNull { text.contains(it) } != null && (instruction == null && !parentHasInstruction)) {
+                context.report(
+                    MISSING_INSTRUCTION, node, location = context.getNameLocation(node),
+                    "Missing instructions attribute"
+                )
+            }
+        }
+
+        if (nodeType == Node.ELEMENT_NODE && node.nodeName == "plurals") {
+            val instruction = node.attributes.getNamedItem("instruction")
+            val translatable = node.attributes.getNamedItem("translatable")
+            val translatableValue = (translatable as Attr?)?.value
+
+            parentInstruction = (translatableValue != "false") && instruction != null
+        }
+
+        // Visit children
+        val childNodes = node.childNodes
+        var i = 0
+        val n = childNodes.length
+        while (i < n) {
+            val child = childNodes.item(i)
+            visitNode(context, child, parentInstruction)
+            i++
+        }
+    }
+
+
+    companion object {
+        val PLACEHOLDERS = listOf(
+            "%s", "%d", "%1\$"
```

Comment:
are `%s` and `%d` still required here given your other lint rule?

Reply:
I'm going to say yes because the idea is to make sure that if a placeholder exists we require instructions and all of those are placeholders. If we were to delete the other rule then we would be missing a few cases.